### ISSUE TITLE
Make a function to create projectors

### ISSUE THREAD

**Issue by joefowler** [usnistgov/Pope.jl#11]
@ggggggggg proposes that I write a function to accept a noise model and a basis to span a subspace of all possible pulses, returning a matrix of projectors. Possible methods:

```
function computeprojectors(basis::AbstractMatrix, noisemodel::ARMA.ARMAModel)
function computeprojectors(basis::AbstractMatrix, noisecovariance::AbstractMatrix)
```

Both methods would take an *n* x *N* matrix `basis` whose *n* columns are the *n* basis elements, each of length *N*. Method 1 would use an approximation to the noise, expressed as a low-order ARMA model. Method 2 would use the estimated noise autocorrelation function (which must be of length *N* or greater). Both would return a 2-tuple with elements:

1. `projectors::Matrix` of size *N* x *n*. This is the transpose of the shape of `basis`, and it makes sense because you could then compute lots of projections in one expression via `projectors*datam`, where `datam` is a matrix whose *N_pulses* columns each represent a single pulse record.
1. `pcovariance::Matrix` of size *n* x *n*, the expected covariance of the projections. This expectation assumes that the noise is additive and Gaussian and independent of the signal level.

The function will *not* assume that `basis` has orthonormal columns. The idea is that `adata = basis*projectors*data` approximates the data vector `data`, in the sense that
* Approximation `adata` is in the space spanned by `basis` (by construction), and
* No other vector in that space is closer to `data`, with "closer" measured by Mahalonobis (signal-to-noise) distance rather than Euclidean distance.

For times when you prefer Euclidean distance and want to ignore the `pcovariance` result, then we could add a third method
```
computeprojectors(basis::AbstractMatrix) = (pinv(basis), eye(size(basis)[2])
```
Is it okay to return a different tuple for this method? If so, then we could omit the second return value.

**Comment by joefowler** [comment]
Tests:
1. Error if noisecovariance argument is too short.
1. Error if basis columns aren't independent.
1. Error if noise is zero.
1. Error if noise is ridiculous (makes a non-positive-definite noise covariance matrix).
1. Error if basis has more columns than rows.
1. Verify `projectors*basis == eye(...)`
1. Verify both methods give same projectors if noisecovariance is that of the ARMA model.
1. Verify projectors are the same if you scale the columns of the basis, but pcovariance changes.
1. Verify projectors are the same if you scale the noise model, but pcovariance changes.

**Comment by joefowler** [comment]
Fixed by #12, but we apparently forgot to close this issue.
